Authenticator/Source/OTPAppDelegate.swift

```diff
@@ -80,3 +80,53 @@ class OTPAppDelegate: UIResponder, UIApplicationDelegate {
         return false
     }
 }
+
+extension OTPAppDelegate: MasterPresenter {
+    func beginAddToken() {
+        if QRScanner.deviceCanScan {
+            let scannerViewController = TokenScannerViewController() { [weak self] (event) in
+                switch event {
+                case .Save(let token):
+                    self?.tokenManager.addToken(token)
+                case .Close:
+                    self?.dismissViewController()
+                }
+            }
+            presentViewController(scannerViewController)
+        } else {
+            let form = TokenEntryForm() { [weak self] (event) in
+                switch event {
+                case .Save(let token):
+                    self?.tokenManager.addToken(token)
+                case .Close:
+                    self?.dismissViewController()
+                }
+            }
+            let formController = TokenFormViewController(form: form)
+            presentViewController(formController)
+        }
+    }
+
+    func beginEditPersistentToken(persistentToken: PersistentToken) {
+        let form = TokenEditForm(token: persistentToken.token) { [weak self] (event) in
+            switch event {
+            case .Save(let token):
+                self?.tokenManager.saveToken(token, toPersistentToken: persistentToken)
+            case .Close:
+                self?.dismissViewController()
+            }
+        }
+        let editController = TokenFormViewController(form: form)
+        presentViewController(editController)
+    }
+
+    private func presentViewController(viewController: UIViewController) {
+        let navController = UINavigationController(rootViewController: viewController)
+        navController.navigationBar.translucent = false
+        window?.rootViewController?.presentViewController(navController, animated: true, completion: nil)
```

Comment:
Line Length Violation: Line should be 100 characters or less: currently 105 characters (line_length)